### DISCUSSION TITLE
[Design] Groups in V3

### DISCUSSION THREAD

**Discussion by lpatmo** [codebuddies/frontend#142] [Ideas]
I've been thinking more and more that instead of what we currently have on codebuddies.org/groups, v3 groups (which to be fair are further off in the future) when we get to it should have an explicit start/end date. And if the group dies out in the middle of week 4 out of 12 of accomplishing its goal, that's fine. But it should end at some date, and not continue forever, because that gives rise to increased risks where a new person eagerly joins a group but it hasn't been active for some time.

**Comment by lpatmo** [comment]
... and by that same token I can imagine multiple cohorts of study groups starting at different times.

**Comment by BethanyG** [comment]
I think that would be good.  I think the only thing I am "spinning" or "noodling" on is the _mechanism_.  Having done battle with channitor in Slack, it doesn't feel good to have an "Auto-expire" based on inactivity.  Some groups may be more leisurely or sporadic than a sharp time-period cutoff.....which is funny coming from me, because I am the queen of nuking due to inactivity.  But having seen people come back after several months or longer break to start up studying again, I think it does make sense to give people some leeway.  

I hard agree that we need to have a solid mechanism to flag/mark/deactivate truly zombie groups so that users aren't frustrated with inactivity (and we're not frustrated with a bunch of outdated info cluttering up the place).

But I also am wondering **what role groups play** in this new system.  If groupings are going to form spontaneously around resources or subjects, and users have the option of connecting directly and continuing that relationship, why would they also want more general groups?  Would that be to declare interest?  To form a support network that was beyond or past the hangout metaphor?  Are groups where discussion threads and collaborative notetaking happen??  Are they connected to learning paths?  (to push even further into the future).  What's a group _for_?

**Reply by lpatmo** [reply]
The more I think about this, the more I wonder if I really am talking about two different types of groups:

1/ forever group (kinda like Slack channels and what already exists on codebuddies.org/study-groups) -- but the downside is admins will have to make sure to minimize duplicates, which Slack channels are also wont to be. And then there are also details to be figured out around what people _do_ in a group -- get notifications about hangouts? announcements? (we also have those features on codebuddies.org right now, but the entire feature has not generated enough activity)

2/ "study group" with an explicit start and end time, and anyone who misses it will have to organize a new cohort. This also assumes certain responsibilities for the leader(s) of each study group. Pro: clearly defined group size and start/end times means there is higher chance of activity. 

Maybe one feature of groups is to  _have_ study groups with different cohorts starting at different times?

(Again, far off in the future, so just brainstorming)

**Reply by BethanyG** [reply]
🤔   So _topics_ vs _study groups_.  So then maybe _groups_ are downsized the way you say in point 2...and maybe those "forever" groups are more a function of _discussion topics_ or _topics_ that people can subscribe to??  Humm....again, brainstorming.  Have to do more thinking on it.
